### DISCUSSION TITLE
Sort function when develop smartcontract

### DISCUSSION THREAD

**Discussion by thaopt** [OpShin/opshin#379] [Ideas / Improvements]
- I want to create uniq asset name when minting a token by hashing the input ref, but it need to sort the oref follwing an order => need to sorted the input.
- currently unable to use sorted function inside validator, any alternative?
- i've tried implement the quicksort myself, but cant swap two item within a list, any advice

**Comment by thaopt** [comment]
Please give me advice or sample if you have, thanks @nielstron 
Do you have any direct contact channel so i can contact you if faced any problem?

**Comment by nielstron** [comment]
I recommend you don't actually sort inside the validator (too expensive!) but expect the caller to pass in a sorted list. you can then just check that subsequent elements are <=

if you really want to sort, you need to implement something that works on immutable lists, i.e. not in-place like [merge sort](https://en.m.wikipedia.org/wiki/Merge_sort)

**Comment by thaopt** [comment]
oke, thanks. i know how it cost but sometime still need to do that while checking logic. haskell have built in sort function, just ask to be sure 'bout that :p
